packages/react-components/react-jsx-runtime/src/jsx.ts

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import {
+  SlotRenderFunction,
+  UnknownSlotProps,
+  slotInternalChildrenSymbol,
+  slotExternalChildrenSymbol,
+} from '@fluentui/react-utilities';
+
+/**
+ * Equivalent to React.createElement but supporting v9 slot API
+ */
+export function jsx<P extends {}>(
+  type: React.ElementType<P>,
+  props?: P | null,
+  ...children: React.ReactNode[]
+): React.ReactElement<P> | null {
+  return isSlotComponent(type)
+    ? jsxFromSlotComponent(type, props, ...children)
+    : React.createElement(type, props, ...children);
+}
+
+function jsxFromSlotComponent<Props extends UnknownSlotProps>(
+  component: React.ElementType<Props>,
+  props?: Props | null,
+  ...overrideChildren: React.ReactNode[]
+): React.ReactElement<Props> | null {
+  const children = normalizeChildren(props, overrideChildren);
+  const {
+    [slotInternalChildrenSymbol as keyof Props]: _,
+    [slotExternalChildrenSymbol as keyof Props]: externalChildren,
+    ...propsWithoutSymbols
+  } = props ?? {};
+
+  if (typeof externalChildren === 'function') {
+    const render = externalChildren as SlotRenderFunction<Props>;
+    return React.createElement(
+      React.Fragment,
+      {},
+      render(component, { ...propsWithoutSymbols, children } as Omit<Props, 'as'>),
+    ) as React.ReactElement<Props>;
+  }
+
+  return React.createElement<Props>(component, { ...propsWithoutSymbols, children } as Props);
+}
+
+export function extractChildrenFromProps<Props extends React.PropsWithChildren<{}>>(
+  props?: Props | null,
+): React.ReactNode[] {
+  if (!props) {
+    return [];
+  }
+  return Array.isArray(props.children) ? props.children : [props.children];
+}
+
+function normalizeChildren<Props extends UnknownSlotProps>(
+  props?: Props | null,
+  overrideChildren: React.ReactNode[] = [],
+): React.ReactNode | undefined {
+  const internalChildren = props && (props[slotInternalChildrenSymbol as keyof Props] as React.ReactNode);
+  const externalChildren =
+    props && (props[slotExternalChildrenSymbol as keyof Props] as React.ReactNode | SlotRenderFunction<Props>);
+  const children: React.ReactNode | React.ReactNode[] =
+    (overrideChildren.length > 0 ? overrideChildren : props?.children) ??
+    (typeof externalChildren === 'function' ? internalChildren : externalChildren) ??
+    internalChildren;
+  return Array.isArray(children) ? React.createElement(React.Fragment, {}, ...children) : children;
+}
+
+export function isSlotComponent<P extends {}>(type: React.ElementType<P>, props?: P | null) {
```

Comment:
```suggestion
export function isSlotComponent<P extends {}>(props: P | null) {
```

Do we need `type` there?

Reply:
No, we don't